x-pack/plugins/lens/public/editor_frame_service/editor_frame/state_helpers.ts

```diff
@@ -104,8 +104,18 @@ export const validateDatasourceAndVisualization = (
       longMessage: string;
     }>
   | undefined => {
+  const layersGroups =
+    currentVisualizationState &&
+    currentVisualization?.getLayerIds(currentVisualizationState).map((layerId) => {
```

Comment:
This has to be a map (with the keys being the layer id) - otherwise the datasource won't be able to map the right group arrays back to the right internal layer representation.

The type should be
```
getErrorMessages: (
    state: T,
    layersGroups?: Record<string, VisualizationDimensionGroupConfig[]>
  ) => Array<{ shortMessage: string; longMessage: string }> | undefined;
```

Reply:
It is an array, so always in order, so it shouldn't be a problem: probably the implementation will need to identify layers index anyway internally to output a better contextual message.
I understand it may be also useful to have a lookup by `layerId`, I'll update.